tests/Feature/ApiUserProviderGuardTest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Amin3536\LaravelApiUserProvider\Tests\Feature;
+
+use Amin3536\LaravelApiUserProvider\authService\ExternalUserProvider;
+use Amin3536\LaravelApiUserProvider\interactModule\HttpClient;
+use Amin3536\LaravelApiUserProvider\Tests\TestCase;
+use Illuminate\Foundation\Auth\User;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Str;
+use Mockery\MockInterface;
+
+class ApiUserProviderGuardTest extends TestCase
+{
+    /**
+     * Test unauthorized request.
+     *
+     * @return void
+     */
+    public function test_unauthorized_request()
+    {
+        $response = $this->getJson(route('user'));
+
+        $response->assertStatus(Response::HTTP_UNAUTHORIZED);
+    }
+
+    /**
+     * Test handle auth server return server error(500).
+     *
+     * @return void
+     */
+    public function test_auth_server_return_server_error()
+    {
+        $mockHttpClient = $this->partialMock(HttpClient::class, function (MockInterface $mock) {
+            $mock->shouldReceive('createRequest')->andReturnSelf();
+
+            $mock->shouldReceive('sendRequest')
+                ->andReturn(new Response([], Response::HTTP_INTERNAL_SERVER_ERROR));
```

Comment:
sendRequest return type is Guzzle response interface please fix it  .
and check how these tests passed! . check it in the userProvider class